core/src/commonMain/kotlin/com/xebia/functional/xef/auto/AIRuntime.kt

```diff
@@ -13,9 +15,14 @@ data class AIRuntime<A>(val runtime: suspend (block: AI<A>) -> A) {
     @OptIn(ExperimentalTime::class)
     @JvmStatic
     fun <A> openAI(): AIRuntime<A> = AIRuntime { block ->
-      val openAIConfig = OpenAIConfig()
-      KtorOpenAIClient(openAIConfig).use { openAiClient ->
-        val embeddings = OpenAIEmbeddings(openAIConfig, openAiClient)
+      val openAIConfig =
+        OpenAIConfig(
+          token =
+            requireNotNull(getenv("OPENAI_TOKEN")) { "OpenAI Token missing from environment." },
+        )
+      val openAI = OpenAI(openAIConfig)
+      OpenAIClient(openAI).use { openAiClient ->
```

Comment:
This should probably be parameterised in the constructor of `openAI`, since not everyone might want to use environment variables for this. Perhaps it's coming from a `application.conf` instead.

Reply:
I parameterized the config as a nullable argument so you can pass it through or, if missing, attempts to load it from the environment in https://github.com/xebia-functional/xef/pull/196/commits/dd47c872288c2b1d23393e5a248f7d146f6071a4